### PR TITLE
Fix Kafka SmolRuntime hang, fixes #532

### LINKED ISSUE #532: Investigate kafka egress issue
Get details on the issue

### PR DESCRIPTION
**Pull request by Licenser**
# Pull request

Fix hang on kafka producer

## Description

This fixes a hang on the kafka producer when using `queue.buffering.max.ms`

## Related

* Related Issues: fixes #532 

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested (we currently can not test Kafka sinks/sources, manual tests will be done instead)
* [x] Use of unsafe code is reasoned about in a comment

### CODE REVIEW

File: src/source/kafka.rs [hunk 1]
```diff
@@ -43,11 +43,11 @@ impl AsyncRuntime for SmolRuntime {
     where
         T: Future<Output = ()> + Send + 'static,
     {
-        async_std::task::spawn(task);
+        smol::spawn(task).detach()
     }
 
     fn delay_for(duration: Duration) -> Self::Delay {
-        async_io::Timer::after(duration).map(|_| ())
+        smol::Timer::after(duration).map(|_| ())
```
Comment (darach) [inline src/source/kafka.rs:50]: Defo needs a comment here - this is so subtle ...

Comment (Licenser) [reply src/source/kafka.rs:52]: :+1:

**Comment by mfelsche** [comment]
Mr. @Licenser can you enlighten me on why this fixes the hangs we've seen? I am more than curious.

**Comment by mfelsche** [comment]
I sneakily rebased on main

**Review by mfelsche — APPROVED** [review]
Verified on my machine. Good job!

**Review by mfelsche — APPROVED** [review]

**Review by mfelsche — APPROVED** [review]
❤️

**Pull request merged**